### DISCUSSION TITLE
Get SEO Data from the _app.js

### DISCUSSION THREAD

**Discussion by SamuelHadsall** [wpengine/faustjs#1649] [Q&A]
Can I render all of my SEO data from the app.js so it's global to the site instead of passing the SEO component to each wp-template file?

I tested and can see that there is already pageProps that can get the fullHead data which looks like `pageProps?.__TEMPLATE_QUERY_DATA__?.page?.seo?.fullHead`

Is there a better way of doing this?

Maybe do a separate graphql query here using nodeByUri?

**Comment by blakewilson** [comment]
@SamuelHadsall If you are certain every page will always request the SEO data or you have proper fallbacks, I don't see why you couldn't read data from the `pageProps` for your `SEO` component. That should work assuming you are always server side rendering your pages.
